### PR TITLE
bitnami/haproxy - loadBalancerSourceRanges should be list and not string

### PR DESCRIPTION
**Pull request by nirroz93**
### Description of the change
Bug fix, using values:
```yaml
service:
  type: LoadBalancer
  loadBalancerSourceRanges: 
    - 1.1.1.1/32
    - 2.2.2.2/32
```
The output template is:
```yaml
# Source: haproxy/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: my-release-haproxy
  namespace: "my-namespace"
  labels:
    app.kubernetes.io/instance: my-release
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: haproxy
    app.kubernetes.io/version: 2.9.0
    helm.sh/chart: haproxy-0.10.9
    app.kubernetes.io/component: haproxy
spec:
  type: LoadBalancer
  externalTrafficPolicy: "Cluster"
  loadBalancerSourceRanges: [1.1.1.1/32 2.2.2.2/32]
```
Which is invalid service

### Benefits
Bug fix

### Applicable issues

- fixes #

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by nirroz93** [comment]
Before the change I'm getting an error:
```Error: UPGRADE FAILED: cannot patch "my-release-haproxy" with kind Service: Service "my-release-haproxy" is invalid: spec.LoadBalancerSourceRanges: Invalid value: "[1.1.1.1/32 2.2.2.2/32]": must be a list of IP ranges. For example, 10.240.0.0/24,10.250.0.0/24 ```
After the change it's working

**Comment by nirroz93** [comment]
The funny thing is that the output of helm template looks the same actually...

File: bitnami/haproxy/templates/service.yaml [hunk 1]
```diff
@@ -25,7 +25,8 @@ spec:
   externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy | quote }}
   {{- end }}
   {{- if eq .Values.service.type "LoadBalancer" }}
-  loadBalancerSourceRanges: {{ .Values.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges:   
+    {{ toYaml .Values.service.loadBalancerSourceRanges | nindent 4 }}
```
Comment (migruiz4) [inline bitnami/haproxy/templates/service.yaml:29]: Could you please keep it in the same line? The `nindent` already adds the new line when rendering

Additionally, please add a condition for empty values:

```suggestion
  {{- if not (empty .Values.service.loadBalancerSourceRanges) }}
  loadBalancerSourceRanges: {{- toYaml .Values.service.loadBalancerSourceRanges | nindent 4 }}
  {{- end }}
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @nirroz93,

Thank you very much for your contribution! Could you please take a look at my comment?

**Comment by nirroz93** [comment]
Thanks @migruiz4 , done

**Review by migruiz4 — APPROVED** [review]
Looks good to me, thank you for your contribution!

**Pull request merged**